### PR TITLE
Bump Rubocop's `TargetRubyVersion` to 2.6 and fix all offenses

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] ~Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).~
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- This was set to 2.3 because Homebrew was running Ruby 2.3.7. Since
  https://github.com/Homebrew/brew/pull/6556, we're running 2.6.3. With
  this change, Rubocop will check against modern Ruby syntax.
- You probably want to review this either commit by commit (with the
  first commit and last two commits being the most important IMO), or with
  `?w=1` as a query string on the diff URL.

### CODE REVIEW

**Comment by issyl0** [comment]
The tests are :green_heart: and so are the style checks. :raised_hands:

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -246,7 +246,7 @@ def check_download_url_format
     end
 
     def bad_url_format?(regex, valid_formats_array)
-      return false unless cask.url.to_s =~ regex
+      return false unless cask.url.to_s&.match?(regex)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb]: `&` isn't needed here.

File: Library/Homebrew/cleanup.rb [hunk 2]
```diff
@@ -400,7 +400,7 @@ def prune_prefix_symlinks_and_directories
           path.extend(ObserverPathnameExtension)
           if path.symlink?
             unless path.resolved_path_exists?
-              if path.to_s =~ Keg::INFOFILE_RX
+              if Keg::INFOFILE_RX.match?(path.to_s)
```
Comment (reitermarkus) [inline Library/Homebrew/cleanup.rb]: The other way around reads nicer I think.

```
path.to_s.match?(Keg::INFOFILE_RX)
```

File: Library/Homebrew/cask/artifact/abstract_uninstall.rb [hunk 3]
```diff
@@ -91,7 +91,7 @@ def uninstall_launchctl(*services, command: nil, **_)
               args: ["list", service],
               sudo: with_sudo, print_stderr: false
             ).stdout
-            if plist_status =~ /^\{/
+            if /^\{/.match?(plist_status)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/abstract_uninstall.rb]: Also switch the order here.

File: Library/Homebrew/dependency_collector.rb [hunk 4]
```diff
@@ -109,7 +109,7 @@ def parse_spec(spec, tags)
   end
 
   def parse_string_spec(spec, tags)
-    if spec =~ HOMEBREW_TAP_FORMULA_REGEX
+    if HOMEBREW_TAP_FORMULA_REGEX.match?(spec)
```
Comment (reitermarkus) [inline Library/Homebrew/dependency_collector.rb]: Also switch the order here.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 5]
```diff
@@ -264,7 +264,7 @@ def audit_file
 
       problem "'__END__' was found, but 'DATA' is not used" if text.end? && !text.data?
 
-      if text =~ /inreplace [^\n]* do [^\n]*\n[^\n]*\.gsub![^\n]*\n\ *end/m
+      if /inreplace [^\n]* do [^\n]*\n[^\n]*\.gsub![^\n]*\n\ *end/m.match?(text.to_s)
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: Also switch the order here.

**Comment by reitermarkus** [comment]
Please go through all the `.match?` statements. I think having the regex second reads a bit nicer.

Also, is portable Ruby 2.6.3 working now on Linux?

File: Library/Homebrew/cask/audit.rb (hunk 1 above)
Comment (issyl0) [reply Library/Homebrew/cask/audit.rb]: Clearly Rubocop's autocorrection is fallible. :-) I'll fix them all.

**Comment by issyl0** [comment]
> Also, is portable Ruby 2.6.3 working now on Linux?

From reading https://github.com/Homebrew/brew/issues/6579, the tests are still disabled, but portable-ruby itself seems to work fine: my local Homebrew has upgraded its Ruby to 2.6.3 and things still work.

**Comment by issyl0** [comment]
I just changed all occurrences of `regexp.match?(string)` to `string.match?(regexp)`, and ow my wrist hurts now, but now I'm getting `undefined method match? for nil:nilClass`. Maybe the [Ruby docs were right that `match?` has to be written regexp first](https://ruby-doc.org/core-2.6.3/Regexp.html#method-i-match-3F), which is why Rubocop auto-corrected in this way in the first place?

**Comment by issyl0** [comment]
Nope, either syntax works, we were calling `.match?` on nil things. This is guarded against by `&.match?` in some places, which I've reinstated. Must have been human error in the swapping. :flushed:

**Comment by issyl0** [comment]
Thanks for the review, @reitermarkus! I've dealt with all your comments.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 6]
```diff
@@ -890,7 +888,7 @@ def audit_text
       end
       bin_names.each do |name|
         ["system", "shell_output", "pipe_output"].each do |cmd|
-          if text =~ /test do.*#{cmd}[\(\s]+['"]#{Regexp.escape(name)}[\s'"]/m
+          if /test do.*#{cmd}[\(\s]+['"]#{Regexp.escape(name)}[\s'"]/m.match?(text.to_s)
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: Missed this one.

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 7]
```diff
@@ -310,7 +310,7 @@ def bump_formula_pr
           end
           username = response.fetch("owner").fetch("login")
         rescue GitHub::AuthenticationFailedError => e
-          raise unless e.github_message =~ /forking is disabled/
+          raise unless /forking is disabled/.match?(e.github_message)
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: And this one.

File: Library/Homebrew/dev-cmd/man.rb [hunk 8]
```diff
@@ -203,7 +203,7 @@ def cmd_comment_manpage_lines(cmd_path)
       end
 
       # Omit the common global_options documented separately in the man page.
-      next if line =~ /--(debug|force|help|quiet|verbose) /
+      next if /--(debug|force|help|quiet|verbose) /.match?(line)
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/man.rb]: And this one.

File: Library/Homebrew/download_strategy.rb [hunk 9]
```diff
@@ -242,7 +242,7 @@ def resolved_url_and_basename
   end
 
   def parse_basename(url)
-    uri_path = if URI::DEFAULT_PARSER.make_regexp =~ url
+    uri_path = if URI::DEFAULT_PARSER.make_regexp&.match?(url)
```
Comment (reitermarkus) [inline Library/Homebrew/download_strategy.rb]: Also this.

**Comment by issyl0** [comment]
Oops, sorry! These should be all fixed now. :male_detective: :female_detective:

File: Library/Homebrew/utils/fork.rb [hunk 10]
```diff
@@ -32,31 +32,29 @@ def self.safe_fork(&_block)
...
+          write.puts error_hash.to_json
+          write.close
 
-            exit!
-          else
-            exit!(true)
-          end
+          exit!
+              else
+                exit!(true)
```
Comment (reitermarkus) [inline Library/Homebrew/utils/fork.rb]: Indentation seems off here.

**Comment by issyl0** [comment]
Thanks for fixing the indentation, @MikeMcQuaid.

File: Library/Homebrew/formula_installer.rb [hunk 11]
```diff
@@ -369,8 +369,8 @@ def check_conflicts
 
       $stderr.puts "Please report this to the #{formula.tap} tap!"
       false
-                else
-                  f.linked_keg.exist? && f.opt_prefix.exist?
+    else # rubocop:disable Lint/ElseAlignment
```
Comment (reitermarkus) [inline Library/Homebrew/formula_installer.rb]: This should probably be reported as a bug.

Comment (reitermarkus) [reply Library/Homebrew/formula_installer.rb]: https://github.com/rubocop-hq/rubocop/issues/6557

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_installer.rb]: Thanks @reitermarkus 🙇

**Pull request merged**